plan/physical_plan_builder.go

```diff
@@ -830,7 +796,8 @@ func (p *Selection) getUsableIndicesAndPk(ds *DataSource) ([]*model.IndexInfo, m
 			continue
 		}
 		cond := pushDownNot(expr.Clone(), false, nil)
-		newConds = append(newConds, substituteCorCol2Constant(cond))
+		newCond, _ := expression.SubstituteCorCol2Constant(cond, false)
```

Comment:
Just return expression.One is easier to give every correlated column a value i think.